### DISCUSSION TITLE
Unable to call component method. Public method [on] not found on component

### DISCUSSION THREAD

**Discussion by magarrent** [livewire/livewire#6061] [7. Bugs (archived)]
### Livewire version

3.0.0-beta-4

### Which PHP version are you using?

PHP 8.1

### Additional information

Hey! Just upgraded from V2 to V3, and it shows me this error on every Livewire component of the application:

`Unable to call component method. Public method [on] not found on component`

The "fix" is simple, just added an **on** method inside the component, but I don't have any idea why this happens.

Any ideas? I know it's difficult to reproduce, but maybe someone knows something :)

[https://flareapp.io/share/bP9pbWom](https://flareapp.io/share/bP9pbWom)

This is the default Jetstream Profile page, and it happens too

**Comment by stevemo** [comment]
I had the same error, found out I was using $wire.on in js script, it must be replace by $wire.$on

Hope this help.

**Comment by magarrent** [comment]
> I had the same error, found out I was using $wire.on in js script, it must be replace by $wire.$on
> 
> Hope this help.


Wow just found it.
![image](https://github.com/livewire/livewire/assets/6561770/b3b22032-63b9-4750-97f4-5c01e71089d5)

Using Jetstream! Just commented for now

**Reply by ciebrant** [reply]
I'm also having the same problem with a fresh Laravel install with Jetstream Livewire... And after update the profile page is giving a error.

**Comment by magarrent** [comment]
Laravel jetstream issues, does not support Livewire v3 at the moment
